packages/ckeditor5-ui/src/editorui/accessibilityhelp.ts

```diff
@@ -0,0 +1,382 @@
+export const DEFAULT_GROUP_ID = 'common' as const;
+
+/**
+ * A plugin that brings the accessibility help dialog to the editor available under the <kbd>Alt</kbd>+<kbd>0</kbd>
+ * keystroke and via the "Accessibility help" toolbar button.
+ */
+export default class AccessibilityHelp extends Plugin {
+	/**
+	 * The view that displays the dialog content (list of keystrokes).
+	 * Created when the dialog is opened for the first time.
+	 */
+	public contentView: AccessibilityHelpContentView | null = null;
+
+	/**
+	 * Keystroke categories, groups and descriptions registered via ({@link #registerKeystrokeCategory}) and
+	 * {@link #registerKeystrokes}.
+	 */
+	private _keystrokes = new Map<string, AccessibilityHelpKeystrokesCategory>();
+
+	/**
+	 * @inheritDoc
+	 */
+	public static get requires() {
+		return [ Dialog ] as const;
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	public static get pluginName() {
+		return 'AccessibilityHelp' as const;
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	public init(): void {
+		const editor = this.editor;
+		const t = editor.locale.t;
+
+		this.registerKeystrokeCategory( DEFAULT_CATEGORY_ID, {
+			label: t( 'Content editing keystrokes' ),
+			description: t( 'These keyboard shortcuts allow for quick access to content editing features.' )
+		} );
+
+		this.registerKeystrokeCategory( 'navigation', {
+			label: t( 'User interface and content navigation keystrokes' ),
+			description: t( 'Use the following keystrokes for more efficient navigation in the CKEditor 5 user interface.' )
+		} );
+
+		this.registerKeystrokes( {
+			category: 'navigation',
+			keystrokes: [
+				{
+					label: t( 'Close contextual balloons, dropdowns, and dialogs' ),
+					keystroke: 'Esc'
+				},
+				{
+					label: t( 'Move focus to the visible contextual balloon' ),
+					keystroke: 'Tab'
+				},
+				{
+					label: t( 'Open the accessibility help dialog' ),
+					keystroke: 'Alt+0'
+				},
+				{
+					label: t( 'Move focus between fields (inputs and buttons) in balloons and dialogs' ),
+					keystroke: 'Tab'
+				},
+				{
+					label: t( 'Move focus to the toolbar, also navigate between toolbars' ),
+					keystroke: 'Alt+F10',
+					mayRequireFn: true
+				},
+				{
+					label: t( 'Navigate through the toolbar' ),
+					keystroke: [ [ 'arrowup' ], [ 'arrowright' ], [ 'arrowdown' ], [ 'arrowleft' ] ]
+				},
+				{
+					label: t( 'Execute the currently focused button' ),
+					keystroke: [ [ 'Enter' ], [ 'Space' ] ]
+				}
+			]
+		} );
+
+		editor.ui.componentFactory.add( 'accessibilityHelp', locale => {
+			const buttonView = new ButtonView( locale );
+
+			buttonView.set( {
+				label: t( 'Accessibility help' ),
+				tooltip: true,
+				withText: true,
+				keystroke: 'Alt+0'
+				// TODO: Hide text, add icon.
+			} );
+
+			buttonView.on( 'execute', () => this._showDialog() );
+
+			return buttonView;
+		} );
+
+		editor.keystrokes.set( 'Alt+0', () => this._showDialog() );
+
+		this._setupRootLabels();
+	}
+
```

Comment:
I think this API is quite messy. On the one hand, we have a method to register categories; on the other hand, we have a method to register groups, keystrokes, or both. Maybe we should provide 3 elementary methods to handle each level separately. So, `registerKeystrokeCategory` should remain as it is. `registerKeystrokeGroup` should be added to handle the options parameter:
```
options: {
    category: string;
    group: string;
    groupLabel: string;
    keystrokes?: Array<AccessibilityHelpKeystrokeDefinition>;
}
```
When keystrokes are provided, the next method to be called will be `registerKeystrokes`. And when it is not necessary to register a new group, we could simply call `registerKeystrokes`.

Reply:
Make sure that it is possible to overwrite a keystroke. I think that when adding a keystroke we could use a label as a key/id.

Reply:
+1 for making it straightforward.

Reply:
F2F decision, let's go with 3 methods.